aws/resource_aws_api_gateway_rest_api.go

```diff
@@ -410,7 +434,8 @@ func flattenApiGatewayEndpointConfiguration(endpointConfiguration *apigateway.En
 	}
 
 	m := map[string]interface{}{
-		"types": flattenStringList(endpointConfiguration.Types),
+		"types":            flattenStringList(endpointConfiguration.Types),
+		"vpc_endpoint_ids": flattenStringList(endpointConfiguration.VpcEndpointIds),
```

Comment:
`vpc_endpoint_ids` shouldn't be added unless the endpoint is of type `PRIVATE` or this will break aws_api_gateway_domain_name